scripts/publish_api_docs_android.sh

```diff
@@ -0,0 +1,161 @@
+        echo "tag ${TAG} exist"
+    else
+        echo "current commit doesn't point at the tag ${TAG}"
+        usage
+    fi
+    echo "VERSION_NUMBER=${VERSION_NUMBER}"
+
+    popd > /dev/null
+}
+
+generateDocs() {
+    pushd "${PATH_TO_REPO}"
+    echo "Start generating docs"
+
+    make javadoc
+
+    echo "Finish generating docs"
+    popd > /dev/null
+}
+
+checkoutToPublisherStaging() {
+    pushd "${PATH_TO_REPO}"
+
+    echo "checkout to ${DOCS_BASE_BRANCH_NAME_STAGING}"
+    git checkout "${DOCS_BASE_BRANCH_NAME_STAGING}"
+
+    popd > /dev/null
+}
+
+checkoutToPublisherProduction() {
+    pushd "${PATH_TO_REPO}"
+
+    echo "checkout to ${DOCS_BASE_BRANCH_NAME_PRODUCTION}"
+    git checkout "${DOCS_BASE_BRANCH_NAME_PRODUCTION}"
+
+    popd > /dev/null
+}
+
+copyDocumentsToDocsFolders() {
+    for fullPath in `find $GENERATED_DOCS_FOLDER -type d -maxdepth 1 -mindepth 1 | grep $DOCS_PREFIX` ; do
+        docsDirectory=`basename $fullPath`
+        targetDirectory=$PATH_TO_REPO/$docsDirectory/$VERSION_NUMBER
+        mkdir $targetDirectory
+        cp -r $fullPath/javadoc/. $targetDirectory
+    done
+}
+
+pushDocsToStaging() {
+    pushd "${PATH_TO_REPO}"
+
+    echo "push docs to staging"
+    git add "./$DOCS_PREFIX*"
+    git commit -m "Docs ${TAG}" -q
+    git push origin HEAD --set-upstream -q
+
+    popd > /dev/null
+}
+
+createBranchProduction() {
+    pushd "${PATH_TO_REPO}"
+
+    echo "create a docs brach"
+    git checkout -b ${BRANCH_NAME}
+    git add "./$DOCS_PREFIX*"
+    git commit -m "Docs ${TAG}" -q
+    git push origin HEAD --set-upstream -q
+    popd > /dev/null
+}
+
+createPullRequest() {
+    pushd "${PATH_TO_REPO}"
+
+    echo "create pull request"
+
+    GITHUB_TOKEN=$GITHUB_PR_ACCESS_TOKEN gh pr create \
+        --title "Docs ${TAG}" \
+        --body "**Docs ${TAG}**
+        <br/>**Staging**:
+        <br/> mapbox-java-core:
+        <a href='https://github.com/mapbox/mapbox-java/tree/publisher-staging/libjava-core/${VERSION_NUMBER}'>staging github</a>
+        <a href='https://docs.tilestream.net/android/java/api/libjava-core/${VERSION_NUMBER}'>staging docs</a>
+        <br/>mapbox-java-geojson:
+        <a href='https://github.com/mapbox/mapbox-java/tree/publisher-staging/libjava-geojson/${VERSION_NUMBER}'>staging github</a>
+        <a href='https://docs.tilestream.net/android/java/api/libjava-geojson/${VERSION_NUMBER}'>staging docs</a>
+        <br/>mapbox-java-turf:
+        <a href='https://github.com/mapbox/mapbox-java/tree/publisher-staging/libjava-turf/${VERSION_NUMBER}'>staging github</a>
+        <a href='https://docs.tilestream.net/android/java/api/libjava-turf/${VERSION_NUMBER}'>staging docs</a>
+        <br/>mapbox-java-services:
+        <a href='https://github.com/mapbox/mapbox-java/tree/publisher-staging/libjava-core/${VERSION_NUMBER}'>staging github</a>
+        <a href='https://docs.tilestream.net/android/java/api/libjava-cores/${VERSION_NUMBER}'>staging docs</a>
```

Comment:
I'm not sure that this is correct URL to staging. Anyway I haven't see working example yet. I asked @colleenmcginnis in slack about it. If stage works I will fix the links

Reply:
https://github.com/mapbox/android-docs/issues/1945

Reply:
Fill fix it in the different PR when find out how stages work

Reply:
More info in https://github.com/mapbox/android-docs/issues/1945#issuecomment-970380071